src/provider/utils/helpers.ts

```diff
@@ -37,6 +37,9 @@ export async function collateClientCalls<T = void>(promises: Promise<T>[]): Prom
         if (successes === 0) {
             result = promiseResult;
         }
+        if (result === undefined && promiseResult !== undefined) {
```

Comment:
Suggest explicitly initialize `result` to `undefined`, and get rid of the `if (successes === 0)` block